src/graphics/pyro/pyro.h

```diff
@@ -185,5 +161,185 @@ class CPyro
     float           m_resetAngle = 0.0f;
 };
 
+class CFlagCreatePyro : public CPyro
+{
+public:
+    CFlagCreatePyro(CObject *pObj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFlagDeletePyro : public CPyro
+{
+public:
+    CFlagDeletePyro(CObject *pObj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CWaypointHitPyro : public CPyro
+{
+public:
+    CWaypointHitPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFallPyro : public CPyro
+{
+public:
+    CFallPyro(CObject *obj);
+
+    //! Start of an object freight falling
+    void        AfterCreate() override;
+    //! Seeks an object to explode by the falling ball of bees
+    CObject*    FallSearchBeeExplo();
```

Comment:
FallSearchBeeExplo() and FallProgress() probably should not be public. Or are they used outside CFallPyro?